File: fooadgram-app/models/content/dto/PostEditDto.ts
import {Visibility} from '../enums/Visibility';
import {RecipeEditDto} from './RecipeEditDto';

export interface PostEditDto {
    title?: string;
    content?: string;
    mediaUrls?: string[];
    tags?: string[];
    visibility?: Visibility;
    location?: string;
    preparationTime?: number;
    recipe?: RecipeEditDto;
} 

File: api/clients/src/main/java/com/foodagram/clients/content/dto/PostEditDto.java
package com.foodagram.clients.content.dto;

import com.foodagram.clients.content.enums.Visibility;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PostEditDto {
    private UUID id;
    private String title;
    private String content;
    private List<String> mediaUrls;
    private List<String> tags;
    private Visibility visibility;
    private String location;
    private RecipeEditDto recipe;
} 
